src/openrct2/world/footpath.c

```diff
@@ -270,7 +270,11 @@ static money32 footpath_element_insert(sint32 type, sint32 x, sint32 y, sint32 z
 
 static money32 footpath_element_update(sint32 x, sint32 y, rct_tile_element *tileElement, sint32 type, sint32 flags, uint8 pathItemType)
 {
-    if (footpath_element_get_type(tileElement) != (type & 0x0F) || (tileElement->type & FOOTPATH_ELEMENT_TYPE_FLAG_IS_QUEUE) != (type >> 7)) {
+    const sint32 newFootpathType = (type & (FOOTPATH_PROPERTIES_TYPE_MASK >> 4));
```

Comment:
This will do for now. Eventually, the signature of `footpath_element_{update,insert}` must be modified.